src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/IllegalTypeCheck.java

```diff
@@ -525,6 +527,19 @@ private boolean isContainVerifiableType(DetailAST modifiers) {
         return result;
     }
 
+    /**
+     * Checks a record component definition. Record components are private
+     * final fields, so we check code in a similar manner as we do for
+     * private final class members.
+     *
+     * @param recordComponentDef the record component definition to check.
+     */
+    private void visitRecordComponentDef(DetailAST recordComponentDef) {
+        if (memberModifiers.isEmpty()) {
```

Comment:
this looks strange.
Lets remove this condition and update javadoc for this property `memberModifiers`. 
from `This property does not affect method calls nor method references. ` to `This property does not affect method calls nor method references nor record components. `

Reply:
@romani sir, I added your condition:
```
    /**
     * Checks a record component definition. Record components are private
     * final fields, so we do not consider memberModifiers field. This
     * property does not affect method calls nor method references
     * nor record components.
     *
     * @param variableDef the record component definition to check.
     */
    private void visitComponentDef(DetailAST variableDef) {
            if (isContainVerifiableType(variableDef)) {
                checkClassName(variableDef);
            }
    }
```

And all my test case for this Issue has been passed BUT I have got one error from another file then this issue:
File Name: InputIllegalTypeRecordsAndCompactCtors.java
Error:
expected   : [27, 31, 32, 35, 39, 40, 48]
but was    : [31, 32, 35, 40, 48]

Should I resolve these errors too? These are not related to this issue.

Reply:
Can we use this condition?
```
    private void visitRecordComponentDef(DetailAST recordComponentDef) {
        if (!memberModifiers.isEmpty()) {
            return;
        }
        checkClassName(recordComponentDef);
    }
```